pkg/ready/ready.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2021 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ready
+
+import (
+	"os"
+	"path/filepath"
+
+	"k8s.io/klog/v2"
+)
+
+// Ready holds state about whether the operator is ready and communicates that
+// to a Kubernetes readiness probe.
+type Ready interface {
+	// Set ensures that future readiness probes will indicate that the operator
+	// is ready.
+	Set() error
+
+	// Unset ensures that future readiness probes will indicate that the
+	// operator is not ready.
+	Unset() error
+}
+
+// NewFileReady returns a Ready that uses the presence of a file on disk to
+// communicate whether a container is ready to start accepting traffic.
+func NewFileReady(component string) Ready {
+	readinessProbeFile := filepath.Join("/tmp", component)
+	return fileReady{
+		FileName: readinessProbeFile,
+	}
+}
+
+type fileReady struct {
+	// FileName represent the full path to the file for determining ready status
+	FileName string
+}
+
+// Set creates a file on disk whose presence can be used by a readiness probe
+// to determine that the operator is ready.
+func (r fileReady) Set() error {
+	f, err := os.Create(r.FileName)
+	if err != nil {
+		if os.IsExist(err) {
+			return nil
+		}
+		return err
+	}
+	klog.Infof("Readiness Probe set to %s", r.FileName)
```

Comment:
Should this be debug (i.e. `V(1)`), wdyt?

Reply:
done